### ISSUE TITLE
Routes issue after updating to 0.29.0

### ISSUE THREAD

**Issue by MDMeridio001** [netbirdio/netbird#2575]
**Describe the problem**

After updating to version 0.29.0 I noticed an issue with some of the routes. If I disconnect and then reconnect a client, some of the routes stop working. This happens for all my clients, the route is added to the OS routing table but the networks remain unreachable. Oddly enough, it doesn't seem to happen whenever the connection to a routing peer is relayed instead of direct. The only workaround I found is disabling and re-enabling the affected routes either from the dashboard or the clients themselves.
I have also noticed that whenever the issue presents itself, the route for the network has a score of 0 as seen in the "client.log" file:

```
New chosen route is crg8ap00o87s73dticd0 with peer aRANLusWN6RNPjrWWXCDEe7F70IElX6/lDSCuC4rIwQ= with score 0.000000 for network [10.6.0.0/24]
```

**To Reproduce**

1. Disconnect and reconnect a client
2. Some of the routes stop working

**Expected behavior**

Network routes shoud be reachable.

**Are you using NetBird Cloud?**

No, self-hosted

**NetBird version**

0.29.0

**NetBird status -dA output:**

Output of one of the clients:
```
Peers detail:
 client2.netbird.selfhosted:
  NetBird IP: 100.66.0.102
  Public key: Yp3lfeq4HN9mGjI/I7SOf6Bpiyv+mwArcY38fI7fQWk=
  Status: Disconnected
  -- detail --
  Connection type:
  ICE candidate (Local/Remote): -/-
  ICE candidate endpoints (Local/Remote): -/-
  Relay server address:
  Last connection update: -
  Last WireGuard handshake: -
  Transfer status (received/sent) 0 B/0 B
  Quantum resistance: false (remote didn't enable quantum resistance)
  Routes: -
  Latency: 0s

 client3.netbird.selfhosted:
  NetBird IP: 100.66.5.21
  Public key: aRANLusWN6RNPjrWWXCDEe7F70IElX6/lDSCuC4rIwQ=
  Status: Connected
  -- detail --
  Connection type: P2P
  ICE candidate (Local/Remote): host/host
  ICE candidate endpoints (Local/Remote): 127.0.0.1:51820/192.168.1.175:51821
  Relay server address: rels://netbird.anon-cU1qT.domain:443
  Last connection update: 13 minutes, 42 seconds ago
  Last WireGuard handshake: 1 minute, 12 seconds ago
  Transfer status (received/sent) 644 B/2.2 KiB
  Quantum resistance: true
  Routes: 10.6.0.0/24
  Latency: 1.7191ms

 client4.netbird.selfhosted:
  NetBird IP: 100.66.41.179
  Public key: 7qIse3SEhrJRl3wpS7X9+woWM4/lM2rD00VAmEasCyM=
  Status: Disconnected
  -- detail --
  Connection type:
  ICE candidate (Local/Remote): -/-
  ICE candidate endpoints (Local/Remote): -/-
  Relay server address:
  Last connection update: -
  Last WireGuard handshake: -
  Transfer status (received/sent) 0 B/0 B
  Quantum resistance: false (remote didn't enable quantum resistance)
  Routes: -
  Latency: 0s

 client5.netbird.selfhosted:
  NetBird IP: 100.66.86.62
  Public key: 6y0DD+Ff3E1Op0FtfWvvX3t3JztaQgsu+T7PxHvkln8=
  Status: Disconnected
  -- detail --
  Connection type:
  ICE candidate (Local/Remote): -/-
  ICE candidate endpoints (Local/Remote): -/-
  Relay server address:
  Last connection update: -
  Last WireGuard handshake: -
  Transfer status (received/sent) 0 B/0 B
  Quantum resistance: false (remote didn't enable quantum resistance)
  Routes: -
  Latency: 0s

 client6.netbird.selfhosted:
  NetBird IP: 100.66.157.250
  Public key: h77qUpO0bhXG6GAzuQ3i43PgLVAP5Wa3Y9Sv7gV37WY=
  Status: Connected
  -- detail --
  Connection type: P2P
  ICE candidate (Local/Remote): host/srflx
  ICE candidate endpoints (Local/Remote): 192.168.117.2:51820/198.51.100.0:51820
  Relay server address: rels://netbird.anon-cU1qT.domain:443
  Last connection update: 12 minutes, 26 seconds ago
  Last WireGuard handshake: 32 seconds ago
  Transfer status (received/sent) 26.9 KiB/18.5 KiB
  Quantum resistance: true
  Routes: 192.168.2.0/24
  Latency: 13.0028ms

OS: windows/amd64
Daemon version: 0.29.0
CLI version: 0.29.0
Management: Connected to https://netbird.anon-cU1qT.domain:443
Signal: Connected to https://netbird.anon-cU1qT.domain:443
Relays:
  [stun:netbird.anon-cU1qT.domain:3478] is Available
  [turn:netbird.anon-cU1qT.domain:3478?transport=udp] is Available
  [rels://netbird.anon-cU1qT.domain:443] is Available
Nameservers:
  [100.66.157.250:53] for [.] is Available
FQDN: client1.netbird.selfhosted
NetBird IP: 100.66.30.225/16
Interface type: Userspace
Quantum resistance: true (permissive)
Routes: -
Peers count: 2/5 Connected
```

**Do you face any (non-mobile) client issues?**

Yes, here is the output of a Windows client but the same thing also happens on Linux:
[netbird.debug.2921127547.zip](https://github.com/user-attachments/files/16950018/netbird.debug.2921127547.zip)

**Comment by moserpjm** [comment]
Same problem here. Router and "client" running Linux.
netbird down, netbird up -> The routes stop working.
Disabling and enabling them again fixes the issue. Also restarting the service helps.
After reading the logs I suspected it has something to do with the new relay feature.

```
2024-09-10T22:12:55+02:00 DEBG [peer: ADVjeYa6yMPg5Lg/Sz+R0Zkh8UoIetI0gtBuRx5YGnw=] client/internal/peer/worker_relay.go:103: enable WireGuard watcher
2024-09-10T22:12:55+02:00 INFO [peer: ADVjeYa6yMPg5Lg/Sz+R0Zkh8UoIetI0gtBuRx5YGnw=] client/internal/peer/conn.go:573: start to communicate with peer via relay
2024-09-10T22:12:55+02:00 DEBG client/internal/routemanager/client.go:190: triggered route state update for Peer ADVjeYa6yMPg5Lg/Sz+R0Zkh8UoIetI0gtBuRx5YGnw=, state: Connected
2024-09-10T22:12:55+02:00 WARN client/internal/routemanager/client.go:130: peer ADVjeYa6yMPg5Lg/Sz+R0Zkh8UoIetI0gtBuRx5YGnw= has 0 latency
2024-09-10T22:12:55+02:00 INFO client/internal/routemanager/client.go:171: New chosen route is co2sqbl5c20s73a1g6kg:cqapjel5c20s738j02tg with peer ADVjeYa6yMPg5Lg/Sz+R0Zkh8UoIetI0gtBuRx5YGnw= with score 0.000000 for network [10.244.0.0/16]
```

Turns out I was right. After removing the relay from management.json everything started working normally again. I also noticed that the clients only forget the relay if I restart the service. IDK if this is by design.

**Comment by mlsmaycon** [comment]
Hey @MDMeridio001 , please test the new 0.29.1 release. You must upgrade your relay and management servers before upgrading the clients if running self-hosted.

**Comment by MDMeridio001** [comment]
@mlsmaycon The issue seems to be resolved completely with version 0.29.1. Thanks!

**Comment by mlsmaycon** [comment]
Great, thanks for the feedback
